plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeConfig.java

```diff
@@ -14,15 +14,22 @@
 package io.trino.plugin.exchange;
 
 import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+import io.airlift.units.MinDataSize;
 
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotNull;
 
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+
 public class FileSystemExchangeConfig
 {
     private String baseDirectory;
     private boolean exchangeEncryptionEnabled;
+    private DataSize maxPageStorageSize = DataSize.of(16, MEGABYTE);
```

Comment:
My consideration here is, if at the end we have incomplete byte ranges for a slice, then we need to copy the bytes to the front of the buffer. If max page storage size is 15MB, and buffer size is 20MB, then that means we can't read 4 chunks in parallel. Therefore it might make sense to be slightly larger than that.